internal/service/userService.go

```diff
@@ -0,0 +1,69 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/internal/repository"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/internal/service/models"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/internal/service/models/exceptions"
+	logger "github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/topology"
+)
+
+type Service interface {
+	SaveUser(ctx context.Context, user models.User) (*models.User, error)
+	GetUserById(ctx context.Context, id string) (*models.User, error)
+}
+
+type service struct {
+	repository repository.Repository
+}
+
+func NewUserService(repository repository.Repository) Service {
+	return &service{
+		repository: repository,
+	}
+}
+
+const errorMessage = "something went wrong"
+
+func (srv *service) SaveUser(ctx context.Context, user models.User) (*models.User, error) {
+	logger.Infof("Saving user %s", user.GetFullName())
+	userByFullName, err := srv.repository.GetUserByFullName(ctx, user.GetFirstName(), user.GetLastName())
+
+	if userByFullName != nil {
+		if err == nil {
+			logger.Errorf("User already exists: %v", user.GetFullName())
+			return nil, nil
+		}
+		logger.Errorf("Something went wrong: %v", err)
+		return nil, &exceptions.DatabaseError{Err: errors.New(errorMessage)}
```

Comment:
We should return the raw error, not assume it's a Database error. See https://github.com/wexinc/ps-tag-payment-scheduler/blob/a6a283834641399513ba31f31bac27fda8aa7279/internal/domain/payment/repo/payment_repo.go#L27-L34

Reply:
It's a best practice rely in our errors instead of external errors that I can't handle after. So instead of returning a library error that I can't handle, I create my own exception and throw it